apps/chef_objects/src/chef_regex.erl

```diff
@@ -100,6 +100,10 @@ regex_for(cookbook_name) ->
 regex_for(environment_name) ->
     generate_regex_msg_tuple(?ANCHOR_REGEX(?NAME_REGEX),
                              <<"Malformed environment name. Must only contain A-Z, a-z, 0-9, _ or -">>);
+regex_for(key_name) ->
+    % This might be the same as nodename -- nodename seems to allow ':' as well
+    generate_regex_msg_tuple(?ANCHOR_REGEX(?NAME_REGEX),
```

Comment:
Fixed this and added coverage for other name checks.  Also changed the messages for other types that accept ":" to include it in the list of permitted characters displayed in the error message.